app/javascript/controllers/audio_upload_controller.js

```diff
@@ -0,0 +1,46 @@
+import { Controller } from "@hotwired/stimulus"
+import morphdom from "morphdom"
+
+export default class extends Controller {
+  connect() {
+    this.bindBeforeFrameRender = this.beforeFrameRender.bind(this)
+    this.element.addEventListener("turbo:before-frame-render", this.bindBeforeFrameRender)
+  }
+
+  disconnect() {
+    this.element.removeEventListener("turbo:before-frame-render", this.bindBeforeFrameRender)
+  }
+
+  // skip some updates/deletes for segment_count changes
+  beforeFrameRender(event) {
+    const childrenOnly = true
+    const onBeforeElUpdated = this.skipSegmentChanges
+    const onBeforeNodeDiscarded = this.skipSegmentChanges
+
+    event.detail.render = (currentElement, newElement) => {
+      if (this.isParentTurboFrame(event.target)) {
+        if (this.isInvalidSegmentCount(event.detail.newFrame)) {
+          morphdom(currentElement, newElement, { childrenOnly, onBeforeElUpdated, onBeforeNodeDiscarded })
+        } else {
+          morphdom(currentElement, newElement, { childrenOnly, onBeforeElUpdated })
+        }
+      } else {
+        morphdom(currentElement, newElement, { childrenOnly })
+      }
+    }
+  }
+
+  // skip changes to segment turbo-frames (they update themselves)
+  skipSegmentChanges(el) {
+    return (el.id || "").match(/episode-form-audio-[0-9]+/) ? false : true
```

Comment:
Where is the `el` being passed in from? The only other two usages of `skipSegmentChanges` don't pass in an element from what we see, so we're not sure how it's evaluating.

Reply:
I'm passing this method (`this.skipSegmentChanges`) in as the `onBeforeElUpdated` and `onBeforeNodeDiscarded` options to morphdom.

If you search for those options in the [morphdom docs](https://www.npmjs.com/package/morphdom) ... you'll see they both get set to functions.  And the functions take in either 1 or 2 elements ... since I don't care about the 2nd element, I just have the 1st one in the method definition here.